recipes-bsp/bootfiles/rpi-config_git.bb

```diff
@@ -45,84 +45,84 @@ do_deploy() {
     cp ${S}/config.txt $CONFIG
 
     if [ -n "${KEY_DECODE_MPG2}" ]; then
-        sed -i '/#decode_MPG2=/ c\decode_MPG2=${KEY_DECODE_MPG2}' $CONFIG
+        echo "decode_MPG2=${KEY_DECODE_MPG2}" >> $CONFIG
```

Comment:
I understand what this does, but the annoyance is that we move context from the sections that we have in the config. Maybe we should have a sed that changes existing lines **or** appends when it isn't found. Maybe an awk?

Reply:
There's no need to force a context whatsoever. If there is an issue with the ordering (like the start_x parameter), this can be explicitly documented, but there is no need to provide arbitrary context in the meantime.

Reply:
Unless you have been using this layer for years and are relying on that information and format.

Reply:
@kraj your thoughts on this?

Reply:
We have run into problems in past when we try to add lines, it could be that bootloader parser is limited in what it can read. I remember one pull where a setting would not take effect because line length was getting longer than 80 chars or some such. So I am a bit nervious to replace sed with echo like this everywhere.

Reply:
> We have run into problems in past when we try to add lines, it could be that bootloader parser is limited in what it can read. I remember one pull where a setting would not take effect because line length was getting longer than 80 chars or some such. So I am a bit nervious to replace sed with echo like this everywhere.

sed doesn't guarantee anything. This is already handled and actually improved in 24984b6... see the commit message.

Reply:
To not block this, what about splitting this into a separate PR?